app/scripts/controllers/client/CreateClientController.js

```diff
@@ -127,6 +192,74 @@
                     this.formData.savingsProductId = null;
                 }
 
+                if(scope.enableAddress===true)
+                {
+                    for(var i=0;i<scope.addressArray.length;i++)
+                    {
+                        var temp=new Object();
+                        if(scope.addressArray[i].addressTypeId)
```

Comment:
Address type Id would made mandatory. There is a column in the m_field_configuration table which marks the filed as mandatory.